module/zfs/dmu.c

```diff
@@ -1767,7 +1770,7 @@ dmu_write_policy(objset_t *os, dnode_t *dn, int level, int wp, zio_prop_t *zp)
 		 * pipeline.
 		 */
 		compress = ZIO_COMPRESS_OFF;
-		checksum = ZIO_CHECKSUM_OFF;
+		checksum = ZIO_CHECKSUM_NOPARITY;
```

Comment:
It will, we can't do this until we port the multi vdev crash feature.  We haven't bothered to port that feature to ZoL because it doesn't benefit the Linux port in any way.  The only reason to do it is to avoid issues like this.